### DISCUSSION TITLE
How to specify which information to output when outputting csv files

### DISCUSSION THREAD

**Discussion by nbxiglk0** [projectdiscovery/httpx#922] [Q&A] [answered]
Hi, When i use httpx to get output as csv file,httpx will use all the default probes, How can I only get the information I specify?  
the command like this  
` cat doamins.txt | httpx   -ip -title -cname -location -random-agent -silent -web-server -status-code -tech-detect -o test.csv -csv`
and the csv file will output all the information.  
![image](https://user-images.githubusercontent.com/24366795/210540977-85330532-600e-4e97-8f9f-1e8e95cd8b42.png)

**Comment by ehsandeep** [comment, marked as answer]
@nbxiglk0 `-csv` follows the same style as `-json` output, i.e all the default fields will be included as part of the output, you will need to do post-processing to filter specific fields.

**Reply by nbxiglk0** [reply]
@ehsandeep Thanks for your answer,
I wonder if it is possible to add this function in future updates to specify which fields to output instead of outputting all fields?

and I just encountered another problem. When entering a url with a protocol or port, the cname flag does not seem to work. Only when the host is entered will the cname record be detected. I wonder if this is originally designed like this? After all, most The host list I entered may contain some other ports, such as 8080, 8443, etc. When I use httpx detection, I also hope to get the cname record of the domain name.
![image](https://user-images.githubusercontent.com/24366795/210760017-6ebcde3b-f5ae-4184-810d-6e45d15394e5.png)

**Reply by ehsandeep** [reply]
@nbxiglk0, you are running an older version of the project, the latest version of httpx is v1.2.5 and does have this issue.

**Reply by nbxiglk0** [reply]
@ehsandeep Thanks for your answer.
